testsuite/openshift/envoy.py

```diff
@@ -2,43 +2,47 @@
 from functools import cached_property
 from importlib import resources
 
+from openshift import Selector
+
 from testsuite.httpx import HttpxBackoffClient
-from testsuite.objects import LifecycleObject
 from testsuite.openshift.client import OpenShiftClient
+from testsuite.openshift.httpbin import Httpbin
+from testsuite.openshift.objects.proxy import Proxy
+from testsuite.openshift.objects.route import OpenshiftRoute, Route
 
 
-class Envoy(LifecycleObject):
+class Envoy(Proxy):
     """Envoy deployed from template"""
-    def __init__(self, openshift: OpenShiftClient, authorino, name, label, httpbin_hostname, image) -> None:
+    def __init__(self, openshift: OpenShiftClient, authorino, name, label, httpbin: Httpbin, image) -> None:
         self.openshift = openshift
         self.authorino = authorino
         self.name = name
         self.label = label
-        self.httpbin_hostname = httpbin_hostname
+        self.httpbin_hostname = httpbin.url
         self.image = image
 
-        self.envoy_objects = None
+        self.envoy_objects: Selector = None  # type: ignore
 
     @cached_property
-    def route(self):
+    def route(self) -> Route:
         """Returns route for object"""
         with self.openshift.context:
             return self.envoy_objects\
                 .narrow("route")\
                 .narrow(lambda route: route.model.metadata.name == self.name)\
-                .object()
+                .object(cls=OpenshiftRoute)
 
-    def create_route(self, name):
```

Comment:
Creating a route is better IMHO. The method that would create a route for Envoy and returns created route. The user can do what ever he wants with that route (e.g. get hostname from it)

Reply:
You can have only one hostname for Route however you can have multiple ones for HTTPRoute